packages/fungible-token/bridge-fungible-token/src/main.sw

```diff
@@ -192,32 +209,46 @@ impl Bridge for Contract {
     }
 
     #[storage(read)]
-    fn asset_to_sub_id(asset_id: b256) -> b256 {
-        _asset_to_sub_id(AssetId::from(asset_id))
+    fn asset_to_sub_id(asset_id: AssetId) -> SubId {
+        _asset_to_sub_id(asset_id)
     }
 }
 
-impl FRC20 for Contract {
+impl SRC20 for Contract {
     #[storage(read)]
-    fn total_supply() -> U256 {
-        let minted: u64 = storage.tokens_minted.try_read().unwrap_or(0);
+    fn total_assets() -> u64 {
+        storage.total_assets.try_read().unwrap_or(0)
+    }
 
-        U256::from((0, 0, 0, minted))
+    #[storage(read)]
+    fn total_supply(asset: AssetId) -> Option<u64> {
+        storage.tokens_minted.get(asset).try_read()
     }
 
     #[storage(read)]
-    fn name() -> str[64] {
-        NAME
+    fn name(asset: AssetId) -> Option<String> {
```

Comment:
For context, let's remember that an instance of this sway contract can only hold one EVM token, 
- with ERC20 tokens using a token_id = sub_id = 0, which translates as having an unique AssetId that can have any amount from 1 to U64.max
- with ERC721 using token_id for each collectible, which translates as having multiple AssetIds that can have only 1 unit each, but still all belong to the same "collection" that is ERC721, and all of these have the same symbol

Illustration:
- ERC20: you are bridging 10 USDT and that translates to a single AssetId say 0x0..abcd, with amount 10. Then, if you query the name for AssetId 0x0..abcd, it will return "Fuel USDT"
- ERC721: you are bridging Ape 1, Ape 2 and Ape 3 with token_id's 1, 2, 3. That translates to AssetId 0x0..1, 0x0..2, 0x0..3, with amount 1 for each one of them. Then, if you query any of these assets for its name, it should return the same name: "Fuel Ape", which is the original ERC721 collection.